cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -966,6 +985,138 @@ public void close(QueryHandle handle) throws ExploreException, HandleNotFoundExc
     return result;
   }
 
+  @Override
+  public void upgrade() throws Exception {
+    // all old CDAP tables used to be in the default database
+    LOG.info("Checking for tables that need upgrade...");
+    List<TableNameInfo> tables = getTables("default");
+    for (TableNameInfo tableNameInfo : tables) {
+      String tableName = tableNameInfo.getTableName();
+      TableInfo tableInfo = getTableInfo(tableNameInfo.getDatabaseName(), tableName);
+      if (!requiresUpgrade(tableInfo)) {
+        continue;
+      }
+
+      String storageHandler = tableInfo.getParameters().get("storage_handler");
+      if (StreamStorageHandler.class.getName().equals(storageHandler)) {
+        LOG.info("Upgrading stream table {}", tableName);
+        upgradeStreamTable(tableInfo);
+      } else if (DatasetStorageHandler.class.getName().equals(storageHandler)) {
+        LOG.info("Upgrading record scannable dataset table {}.", tableName);
+        upgradeRecordScannableTable(tableInfo);
+      } else {
```

Comment:
Its guaranteed to always be a ``FileSet`` if its not one of the above two, right? Is there a Storage handler for Filesets that we could check for here?